runtime/compiler/runtime/J9CodeCacheManager.cpp

```diff
@@ -298,6 +301,18 @@ J9::CodeCacheManager::allocateCodeCacheSegment(size_t segmentSize,
       {
       vmemParams.pageSize = largeCodePageSize;
       vmemParams.pageFlags = config.largeCodePageFlags();
+
+      // Recalculate the page size if using large pages
+      if(vmemParams.pageSize > ccTotalSizeB)
+         {
+         for (UDATA pageIndex = 0; 0 != pageSizes[pageIndex]; ++pageIndex)
```

Comment:
Is the array of page sizes ordered from small pages to larger pages? If so, the code works, otherwise we may have to keep the maximum page size that is smaller than total code cache.

Reply:
Yes it currently populates the array in the increasing order with [0] holding the default of 4K for linux. 
https://github.com/eclipse/omr/blob/15c89a7ec8b03bdc19581be8aa570d0971d60ce5/port/linux/omrvmem.c#L608-L609